### ISSUE TITLE
Crash on Android when async is false and no internet connection

### ISSUE THREAD

**Issue by PaulGene** [yupswing/akifox-asynchttp#30]
Tested with the "Basic example"

It doesn't crash on Windows.

**Comment by yupswing** [comment]
I'll test it as soon as possible today. Thank you for the feedback

**Comment by yupswing** [comment]
I cannot reproduce the bug.
I've tried to set async: false and run the openfl example on android.
No problem at all the normal ERROR: 0

```
12-07 17:57:32.214 17073 17154 I trace   : AsyncHttp.hx:185: uPCnA9fh INFO:Request
12-07 17:57:32.214 17073 17154 I trace   : > GET http://i0.kym-cdn.com/photos/images/list/000/160/616/TROLOLOL.jpg
12-07 17:57:32.224 17073 17154 I trace   : AsyncHttp.hx:194: uPCnA9fh ERROR:Request failed -> Unknown host:i0.kym-cdn.com
12-07 17:57:32.224 17073 17154 I trace   : AsyncHttp.hx:185: uPCnA9fh INFO:Response 0 (0 bytes in 0.005 s)
12-07 17:57:32.224 17073 17154 I trace   : > GET http://i0.kym-cdn.com/photos/images/list/000/160/616/TROLOLOL.jpg
12-07 17:57:32.224 17073 17154 I trace   : Main.hx:64: ERROR (HTTP STATUS 0)
```

Can you give me more details?
At least the target android SDK, and the actual code you have used.

**Comment by PaulGene** [comment]
Ok I just tried your Openfl sample, and with wifi and mobile data off, and async set to false... i get the same crash as before.

When it crashes i get 12-07 21:52:37.706 10026 10057 I trace   : Lib.hx:360: Invalid socket handle

I'm using Haxe 3.3.0 / OpenFL 3.6.1 (Legacy mode) / Lime 2.9.1 / Android SDK = 22

Running on Android 6

**Comment by yupswing** [comment]
I set up my environment to the exact stuff you are using (haxe3.3.0, openfl 3.6.1 (legacy mode), lime 2.9.1, android-sdk=22)

My phone is already running Android 6.

It does still work perfectly, giving ERROR: 0 (http status 0) when wifi and mobile data off.
Have you updated the lib to the last version (0.4.4)?

**Comment by PaulGene** [comment]
It looks like this might be a nightmare to track down as I've just tested on a different device and it works fine.

Acer Liquid E380 + Android 4.4.2 = Never crashes.
Xiaomi Redmi Pro + Android 6.0 = Crashes every single time if no connection present with "Invalid socket handle".

It might be some weird threading issue, the Redmi Pro has a Helio 10-core processor.

Here are my installed libs: http://pastebin.com/wmMwrevE

**Comment by yupswing** [comment]
Ok, help me to find out where the error is thrown.
When async is false in your case it is called httpViaSocket (and httpViaSocketConnect)
Since `.connect` is already in a try, i guess it is not that.
But there are some `.close`, maybe that's it.

Can you tell me the context.
What is print in console before 12-07 21:52:37.706 10026 10057 I trace : Lib.hx:360: Invalid socket handle

Be sure to enable the log with `AsyncHttp.logEnabled = true;` anywhere before the request.

**Comment by PaulGene** [comment]
Ok I did some digging and its crashing with invalid socket handle here https://github.com/yupswing/akifox-asynchttp/blob/master/com/akifox/asynchttp/AsyncHttp.hx#L298

Moving that line into the try block before connect seems to fix it.

**Comment by yupswing** [comment]
brilliant mate, thanks!
fixed on 0.4.6
